test/lit/wat-kitchen-sink.wast

```diff
@@ -1627,6 +1629,22 @@
   array.copy $a2 $a2
  )
 
+ ;; CHECK:      (func $array-fill (type $ref|$a2|_i32_f32_i32_=>_none) (param $0 (ref $a2)) (param $1 i32) (param $2 f32) (param $3 i32)
+ ;; CHECK-NEXT:  (array.fill $a2
+ ;; CHECK-NEXT:   (local.get $0)
+ ;; CHECK-NEXT:   (local.get $1)
+ ;; CHECK-NEXT:   (local.get $2)
+ ;; CHECK-NEXT:   (local.get $3)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $array-fill (param (ref $a2) i32 f32 i32)
+  local.get 0
+  local.get 1
+  local.get 2
+  local.get 3
+  array.fill $a2
+ )
```

Comment:
just `fill` for now and not `init*`?

Reply:
We don't parse element segment references yet, so I left `ArrayInit` as future work.